restrict_download_region/restrict_region.py

```diff
@@ -26,46 +29,24 @@ def handler(event: dict, context: dict):
         if not BUCKET_NAME:
             raise ValueError("BUCKET_NAME must be defined in enviroment variables")
 
-        region_ip_prefixes = get_ip_prefixes_for_region() \
-            if custom_resource_request_type != 'Delete'\
-            else None
-
         s3_client = boto3.client('s3')
 
         # get current bucket_policy from the s3 bucket
         bucket_policy = get_bucket_policy(s3_client, BUCKET_NAME)
 
-        # add/update/remove ip restirction policy depending on whether region_ip_prefixes
-        process_ip_restrict_policy(BUCKET_NAME, region_ip_prefixes,
+        # add/update/remove ip restirction policy depending on the region_ip_prefixes
+        process_ip_restrict_policy(BUCKET_NAME,
                                    custom_resource_request_type, bucket_policy)
 
         # update with newly modified bucket policy
         update_bucket_policy(s3_client, BUCKET_NAME, bucket_policy)
 
 
-def generate_ip_address_policy(bucket_name: str, region_ip_prefixes: List[str]):
-    return {
-        'Sid': POLICY_STATEMENT_ID,
-        'Effect': 'Deny',
-        'Principal': '*',
-        'Action': 's3:GetObject',
-        'Resource': 'arn:aws:s3:::'+bucket_name+'/*',
-        'Condition': {
-            'NotIpAddress': {'aws:SourceIp': region_ip_prefixes},
-            # allows any S3 VPC Endpoint to bypass the ip restriction.
-            # cross region gateway endpoints are not supported in AWS so any S3 VPC endpoint
-            # traffic is implicitly same region.
-            'Null': {'aws:sourceVpc': 'true'}
-        }
-    }
-
-
 def get_ip_prefixes_for_region() -> List[str]:
     if not AWS_REGION:
         raise ValueError("AWS_REGION must be defined in environment variables.")
 
     # generate new policy statement based on data from AWS
-    http = urllib3.PoolManager()
```

Comment:
moved to module level. we technically only call this one time per lambda execution so it doesn't really matter, but `PoolManager()` is supposed to be reusable